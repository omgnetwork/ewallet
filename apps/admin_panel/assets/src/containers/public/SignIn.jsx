import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { Button } from 'react-bootstrap';
import { getTranslate } from 'react-localize-redux';

import { userActions } from "../../actions";
import OMGFieldGroup from "../../components/OMGFieldGroup"

class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
      submitted: false,
      didModifyEmail: false,
      didModifyPassword: false,
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    const { id, value } = e.target;
    var { didModifyEmail, didModifyPassword } = this.state
    if (id === "email") { didModifyEmail = true }
    else if (id === "password") { didModifyPassword = true }
    this.setState({
      [id]: value,
      didModifyEmail: didModifyEmail,
      didModifyPassword: didModifyPassword
    });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.setState({ submitted: true });
    const { email, password } = this.state;
    const { login } = this.props;
    if (email && password) {
      login(email, password)
    }
  }

  isEmailValid() {
    return (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,})+$/.test(this.state.email))
  }

  isPasswordValid() {
    return this.state.password.length >= 8
  }

  isFormValid() {
    return this.isEmailValid() && this.isPasswordValid()
  }

  getEmailValidationState() {
    const { submitted, didModifyEmail } = this.state;
    return (!this.isEmailValid() && (submitted || didModifyEmail)) ? "error" : null
  }

  getPasswordValidationState() {
    const { submitted, didModifyPassword } = this.state;
    return (!this.isPasswordValid() && (submitted || didModifyPassword)) ? "error" : null
  }

  render() {
    const { loading, translate } = this.props;
    const { email, password } = this.state;
    return (
      <div className="row">
        <div className="col-xs-12 col-sm-6 col-sm-offset-3">
          <div className="omg-form">
            <h2 className="omg-form__title">{translate("sign-in.sign-in")}</h2>
            <form onSubmit={this.handleSubmit} autoComplete="off">
              <OMGFieldGroup
                id="email"
                label={translate("sign-in.email.label")}
                help={translate("sign-in.email.help")}
                validationState={this.getEmailValidationState()}
                type="text"
                value={email}
                onChange={this.handleChange}
              />
              <OMGFieldGroup
                id="password"
                label={translate("sign-in.password.label")}
                help={translate("sign-in.password.help")}
                validationState={this.getPasswordValidationState()}
                type="password"
                value={password}
                onChange={this.handleChange}
              />
              <Button bsClass="btn btn-omg-blue"
                bsStyle="primary"
                disabled={loading || !this.isFormValid()}
                type="submit"
              >
                {loading ? translate("global.loading") : translate("sign-in.sign-in")}
              </Button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const { loading } = state.global;
  const translate = getTranslate(state.locale);
  return {
    loading, translate
  };
}

function mapDispatchToProps(dispatch) {
  return {
    login: (email, password) => {
      return dispatch(userActions.login(email, password))
    }
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SignIn));
