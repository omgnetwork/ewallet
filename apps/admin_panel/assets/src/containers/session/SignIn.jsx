import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { Button } from 'react-bootstrap';

import { userActions } from "../../actions";
import OMGFieldGroup from "../../components/OMGFieldGroup"

class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      submitted: false,
      didModifyUsername: false,
      didModifyPassword: false,
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    const { id, value } = e.target;
    var { didModifyUsername, didModifyPassword} = this.state
    if (id === "username") { didModifyUsername = true }
    else if (id === "password") { didModifyPassword = true }
    this.setState({ [id]: value,
                    didModifyUsername: didModifyUsername,
                    didModifyPassword: didModifyPassword
                  });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.setState({ submitted: true });
    const { username, password } = this.state;
    const { dispatch } = this.props;
    if (username && password) {
      dispatch(userActions.login(username, password));
    }
  }

  isUsernameValid() {
    return this.state.username.length >= 3
  }

  isPasswordValid() {
    return this.state.password.length >= 8
  }

  isFormValid() {
    return this.isUsernameValid() && this.isPasswordValid()
  }

  getUsernameValidationState() {
    const { submitted, didModifyUsername } = this.state;
    return (!this.isUsernameValid() && (submitted || didModifyUsername)) ? "error" : null
  }

  getPasswordValidationState() {
    const { submitted, didModifyPassword } = this.state;
    return (!this.isPasswordValid() && (submitted || didModifyPassword)) ? "error" : null
  }

  render() {
    const { loggingIn } = this.props;
    const { username, password } = this.state;
    return (
      <div className="row">
        <div className="col-xs-12 col-sm-8 col-sm-offset-2">
          <div className="omg-form">
            <h2 className="omg-form__title">Sign in</h2>
            <form onSubmit={this.handleSubmit}>
              <OMGFieldGroup
                id="username"
                label="Username"
                help = "Username is required and should be at least 3 characters long"
                validationState={this.getUsernameValidationState()}
                type="text"
                value={username}
                onChange={this.handleChange}
              />
              <OMGFieldGroup
                id="password"
                label="Password"
                help="Password is required and should be at least 8 characters long"
                validationState={this.getPasswordValidationState()}
                type="password"
                value={password}
                onChange={this.handleChange}
              />
              <Button bsClass="btn btn-omg-blue"
                      bsStyle="primary"
                      disabled={loggingIn || !this.isFormValid()}
                      type="submit"
              >
                {loggingIn ? 'Loading...' : 'Sign In'}
              </Button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const { loggingIn } = state.authentication;
  return {
    loggingIn
  };
}

export default withRouter(connect(mapStateToProps)(SignIn));
