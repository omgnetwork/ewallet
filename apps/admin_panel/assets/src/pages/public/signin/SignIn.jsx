import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button } from 'react-bootstrap';
import { withRouter } from 'react-router-dom';
import { getTranslate } from 'react-localize-redux';
import PropTypes from 'prop-types';

import Actions from './actions';
import OMGFieldGroup from '../../../components/OMGFieldGroup';
import OMGLoadingButton from '../../../components/OMGLoadingButton';
import {
  getEmailValidationState,
  getPasswordValidationState,
  isFormValid,
  onInputChange,
  onSubmit,
} from './stateFunctions';

class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      submitted: false, //eslint-disable-line
      didModifyEmail: false, //eslint-disable-line
      didModifyPassword: false, //eslint-disable-line
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    const { target } = e;
    this.setState(prevState => (onInputChange(target, prevState)));
  }

  handleSubmit(e) {
    e.preventDefault();
    this.setState(onSubmit());
    const { email, password } = this.state;
    const { login } = this.props;
    if (email && password) {
      login({ email, password });
    }
  }

  render() {
    const { loading, translate, history } = this.props;
    const { email, password } = this.state;
    return (
      <div className="omg-form">
        <h2 className="omg-form__title">
          {translate('sign-in.sign-in')}
        </h2>
        <form autoComplete="off" onSubmit={this.handleSubmit}>
          <OMGFieldGroup
            help={translate('sign-in.email.help')}
            id="email"
            label={translate('sign-in.email.label')}
            onChange={this.handleChange}
            type="text"
            validationState={getEmailValidationState(this.state)}
            value={email}
          />
          <OMGFieldGroup
            help={translate('sign-in.password.help')}
            id="password"
            label={translate('sign-in.password.label')}
            onChange={this.handleChange}
            type="password"
            validationState={getPasswordValidationState(this.state)}
            value={password}
          />
          <div>
            <span>
              <OMGLoadingButton
                disabled={!isFormValid(this.state)}
                loading={loading}
                type="submit"
              >
                {translate('sign-in.sign-in')}
              </OMGLoadingButton>
              <span className="ml-1">
                {translate('sign-in.or')}
              </span>
              <Button
                bsStyle="link"
                className="link-omg-blue"
                disabled={loading}
                onClick={() => { history.push('/forgot_password'); }}
              >
                {translate('sign-in.forgot_password')}
              </Button>
            </span>
          </div>
        </form>
      </div>
    );
  }
}

SignIn.propTypes = {
  history: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
  login: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
};

function mapStateToProps(state) {
  const { loading } = state.global;
  const translate = getTranslate(state.locale);
  return {
    loading,
    translate,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    login: params => dispatch(Actions.login(params)),
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SignIn));
