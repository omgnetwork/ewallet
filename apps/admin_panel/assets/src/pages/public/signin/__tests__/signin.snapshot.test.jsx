import renderer from 'react-test-renderer';
import SignIn from '../SignIn';
import { SnapshotComponentTester } from '../../../../../__tests__/ComponentTester';

it('renders correctly', () => {
  const tree =
  renderer.create(SnapshotComponentTester(SignIn, { loading: false, login: jest.fn() })).toJSON();
  expect(tree).toMatchSnapshot();
});
