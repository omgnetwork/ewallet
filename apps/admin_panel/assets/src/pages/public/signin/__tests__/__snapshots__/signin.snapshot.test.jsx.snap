// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="omg-form"
>
  <h2
    className="omg-form__title"
  >
    Sign in
  </h2>
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <div
      className="omg-form__group form-group"
    >
      <label
        className="omg-form__label control-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="omg-form__input form-control"
        id="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="omg-form__group form-group"
    >
      <label
        className="omg-form__label control-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="omg-form__input form-control"
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <span>
        <button
          className="btn btn-omg-blue btn btn-omg-blue-primary"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Sign in
        </button>
        <span
          className="ml-1"
        >
          or
        </span>
        <button
          className="link-omg-blue btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Forgot your password?
        </button>
      </span>
    </div>
  </form>
</div>
`;
